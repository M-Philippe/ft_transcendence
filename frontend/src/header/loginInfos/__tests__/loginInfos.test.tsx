import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import LoginInfos from '../loginInfos';
import { Provider } from 'react-redux';

/*
**  'npm test' to run test (and create a snapshot if new test)
**    if a test fails:
**      - it's a bug, so resolve it
**      - it's normal behavior for edited comp (so type u or i to update snapshot).
**
**    if you edit a *.test.tsx file directly, type u after 'npm test'
**    to remove obsolete snapshot
*/

test('<LoginInfos /> display username if bool is set to true', () => {
  const mockStoreConf = configureStore([]);
  
  let mockStore = mockStoreConf({user: {
    username: "pminne",
    isConnected: true,
  }});

  const component = renderer.create(
    <Provider store={mockStore}>
      <LoginInfos />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('<LoginInfos /> don\'t display username if bool is set to false', () => {
  const mockStoreConf = configureStore([]);
  
  let mockStore = mockStoreConf({user: {
    username: "pminne",
    isConnected: false,
  }});

  const component = renderer.create(
    <Provider store={mockStore}>
      <LoginInfos />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})