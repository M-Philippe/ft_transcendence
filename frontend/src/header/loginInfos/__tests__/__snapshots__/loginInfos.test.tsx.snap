// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginInfos /> display username if bool is set to true 1`] = `
<p
  className="login"
>
  Hello 
  pminne
</p>
`;

exports[`<LoginInfos /> don't display username if bool is set to false 1`] = `
<p
  className="login"
>
  No Login
</p>
`;
